site/source/docs/contributing/developers_guide.rst

```diff
@@ -36,37 +57,46 @@ Patches should be submitted as *pull requests* to the **incoming** branch.
-- "Add an automatic test to `tests/runner.py <https://github.com/emscripten-core/emscripten/blob/master/tests/runner.py>`_ if you add any new functionality or fix a bug.
-- Record the tests that were run in the pull request or issue.
-
+- Add an automatic test if you add any new functionality or fix a bug. Search
+  in ``tests/*.py`` for related tests, as often the simplest thing is to add to
+  an existing one. If you're not sure how to test your code, feel free to ask
+  for help.
+- We normally squash and merge PRs, which means the PR turns into a single
+  commit on the target branch. Because of that, it's ok to have merge commits
+  in the PR itself, as they get removed. Please put a good description for
+  the final commit in the PR description, and we'll use it when squashing.
 
 Code reviews
 ============
 
-`@kripken <https://github.com/kripken>`_ reviews all pull requests before merging.
-
-Exceptions are sub-projects that are 'owned' by other people. These owners can push to incoming directly:
-
--  OpenAL and audio in general: `@ehsan <https://github.com/ehsan>`_
--  embind: `@imvu <https://github.com/imvu>`_
--  Windows stuff: `@juj <https://github.com/juj>`_
-
+One of the core developers will review a pull request before merging it. If
+several days pass without any comments on your PR, please comment in the PR
+which will ping them. (If that happens, sorry! Sometimes things get missed.)
 
 Compiler overview
 =================
 
 The :ref:`Emscripten Compiler Frontend (emcc) <emccdoc>` is a python script that manages the entire compilation process:
 
-- *emcc* calls :term:`Clang` to convert C++ to bitcode, ``llvm-opt`` to optimize it, ``llvm-link`` to link it, etc.
-- *emcc* then calls `emscripten.py <https://github.com/emscripten-core/emscripten/blob/master/emscripten.py>`_, which performs any final compilation to wasm (including invoking *wasm-emscripten-finalize* from Binaryen, for wasm) and emits the JavaScript.
-- If optimizing wasm, *emcc* will then call Binaryen's optimizer, run meta-dce, and other useful things. It will also run JS optimizations on the JS that is emitted alongside the wasm.
+- **emcc** calls :term:`Clang` to compile C++ and ``wasm-ld`` to link it. It
+  builds and integrates with the Emscripten system libraries, both the
+  compiled ones and the ones implemented in JS.
+- **emcc** then calls `emscripten.py <https://github.com/emscripten-core/emscripten/blob/master/emscripten.py>`_
+  which performs the final transformation to wasm (including invoking
+  **wasm-emscripten-finalize** from Binaryen) and calls the JS compiler
+  (see ``src/compiler.js`` and related files) which emits the JS.
+- If optimizing wasm, **emcc** will then call **wasm-opt**, run meta-dce, and
```

Comment:
Same for meta-dce (I *think* it was documented somewhere?).

Reply:
there is a high-level description of the entire process here: https://hacks.mozilla.org/2018/01/shrinking-webassembly-and-javascript-code-sizes-in-emscripten/ is that what you mean?

Reply:
Yeah, that looks good as a reference link. It just helps to assume no previous knowledge on parts of the process from the reader IMO.